sunpycube/cube/NDCube.py

```diff
@@ -0,0 +1,220 @@
+from sunpy.visualization.imageanimator import ImageAnimatorWCS
+import matplotlib.pyplot as plt
+import sunpy.visualization.wcsaxes_compat as wcsaxes_compat
+import astropy.units as u
+import sunpycube.wcs_util
+import astropy.nddata
+import numpy as np
+
+__all__ = ['NDCube', 'Cube2D', 'Cube1D']
+
+
+class NDCube(astropy.nddata.NDData):
+    """docstring for NDCube"""
+
+    def __init__(self, data, uncertainty=None, mask=None, wcs=None, meta=None, unit=None, copy=False, missing_axis=None, **kwargs):
+        if missing_axis is None:
+            self.missing_axis = [False]*wcs.naxis
+        else:
+            self.missing_axis = missing_axis
+        if data.ndim is not wcs.naxis:
+            count = 0
+            for bool_ in self.missing_axis:
+                if not bool_:
+                    count += 1
+            if count is not data.ndim:
+                raise ValueError(
+                    "The number of dimensions of data and number of axes of wcs donot match.")
```

Comment:
Message should be "The number of data dimensions and number of wcs non-missing axes do not match."